receiver/nats.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * skogul, nats receiver/subscriber
+ *
+ * Author(s):
+ *  - Niklas Holmstedt <n.holmstedt@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package receiver
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/telenornms/skogul"
+	"sync"
+)
+
+var natsLog = skogul.Logger("receiver", "nats")
+
+/*
+Nats.io simple pub/sub receiver with:
+
+- Authentication: Username/Password, TLS
+- Authorization: Username/Password, UserCredentials/JWT
+- Queue: Load balancing for multiple receivers in the same Queue.
+*/
+type Nats struct {
+	Handler       skogul.HandlerRef `doc:"Handler used to parse, transform and send data."`
+	Servers       string            `doc:"Comma separated list of nats URLs"`
```

Comment:
Use a `[]string` as a list of strings instead of manually comma-separating this to be in line with other Skogul configuration.

If this is how nats accept the configuration, and it is common to use this format, I can accept keeping it this way. Otherwise, I'm more inclined to use `strings.Join()` or something similar to join them back together.

Reply:
Yes, exactly. This is the way all the clients behave, as well as the official cli binaries. Otherwise i probably would have separated them.